### ISSUE TITLE
GEO prefix for Geographical Entity Ontology conflicts with Gene Expression Omnibus

### ISSUE THREAD

**Issue by cthoyt** [ufbmi/geographical-entity-ontology#19]
I've been doing a lot of prefix mapping across various registries and been pleasently surprised to find few conflicts. Unfortunately, both the Geographical Entity Ontology and the Gene Expression Omnibus have used GEO as a prefix. Geographical Entity Ontology is listed in BioPortal, Prefix Commons, OLS, and the OBO Foundry, while Gene Expression Omnibus is listed in Identifiers.org.

However, I think Geographical Entity Ontology was only recently listed in some of these resources. Now it's causing a conflict. Would you consider using a different prefix for your ontology, perhaps `geogeo`?

**Comment by hoganwr** [comment]
I'm not sure I understand.  OBO identifiers are not intended to be globally
unique (although they are when you include the full IRI).  How could they
conflict with other non-OBO resources?

Or is Gene Expression Omnibus somehow now included in OBO scope?

Bill

**Comment by cthoyt** [comment]
It's definitely true that OBO Foundry/OLS aren't enforcing global uniqueness (that would be a pipe dream). It's just by a happy chance that conflicts happen very infrequently with the common prefixes used more generally in the bioinformatics community. Unfortunately, this is one of those cases.

I'll admit I know relatively little about your ontology. Based on its name, I might guess that you're less familiar with bioinformatics resources than semantic web resources.

I'm working on the Bioregistry (https://bioregistry.io/), which is a much larger registry that's combining/normalizing many different ones. Either way, I will have to internally solve the issue of the GEO clash. Since in bioinformatics, GEO always means Gene Expression Omnibus, please let me know under what prefix you would like your ontology listed there (right now, it's `geoeo`)

**Comment by hoganwr** [comment]
I'm not entirely sure what you mean by "prefix".  I mostly understand it to
mean xmlns prefixes.

In that case, we don't use geo as a prefix anywhere in GEO.  All GEO IRIs
have an obo prefix.

I don't fully understand all the consequences of this decision, but I think
it's probably fine to list us under your "prefix" geogeo

Bill

**Comment by cthoyt** [comment]
Sorry to throw around "prefix" as some jargon. I mean the key that's used in OBO Foundry and OLS:

- OBO Foundry: [http://www.obofoundry.org/ontology/**geo**.html](http://www.obofoundry.org/ontology/geo.html)
- OLS: [https://www.ebi.ac.uk/ols/ontologies/**geo**](https://www.ebi.ac.uk/ols/ontologies/geo)

Identifiers.org, which also covers many of the ontologies in OBO Foundry/OLS uses GEO for Gene Expression Omnibus:

- https://registry.identifiers.org/registry/geo

**Comment by hoganwr** [comment]
Thanks for the clarification.  Good information to know.

So as I understand it, your action affects bioregistry but not
identifiers.org, correct?  In which case would it be useful to ensure we
have the same key in both, and thus should try to register geogeo w/
identifiers.org?

**Comment by cthoyt** [comment]
Yeah, that would be a great way to increase visibility of your work! Unfortunately one of the reasons I've been working on the Bioregistry is to make a more community driven version of Identifiers.org. I can't vouch that they will be responsive to requests to add new content

**Comment by hoganwr** [comment]
Fair enough.  Right now, I've got all green checkboxes on the request form,
and I get an error when submitting that says to try again later....

Bill

**Comment by cmungall** [comment]
I would recommend having a more systematic way of allocating namespaces in the case of conflicts. Can we use one of:

 * obogeo
 * geoont

In future, I do not think OBO should grant namespaces that are in use by major databases (https://github.com/OBOFoundry/OBOFoundry.github.io/issues/1522)

**Comment by hoganwr** [comment]
Already applied for and received geogeo

Maybe I could ask them to change

**Comment by cmungall** [comment]
'Them' being identifiers.org?

Yes, they can surely update it.

I think you would want to do this anyway. Whoever registered this made the same mistake that is often made when registering the OBO ontologies of including the prefix in the local part.

https://registry.identifiers.org/registry/geogeo

Your identifiers are "geogeo:GEO_000000021"

I have seen the doubling up before but this is the first time I have seen a triplification. I hope you agree this is undesirable!

I recommend:

 * Prefix `obogeo`
 * URL pattern `http://purl.obolibrary.org/obo/GEO_`
 * Sample URL https://identifiers.org/obogeo:000000021
 * Sample contact ID: obogeo:000000021

**Comment by hoganwr** [comment]
I registered it as a trial run, and I learned a lot.

I used geogeo because it was suggested to me by Charles, who seems to have
much expertise in this area (more than I have, so I certainly didn't see a
reason to question it).

I'll file an update request soon, although iran't really registering an
entirely new prefix?

Bill
